bootable-jar/runtime/src/main/java/org/wildfly/core/jar/runtime/BootableJar.java

```
@@ -0,0 +1,371 @@
+        if (Files.exists(loggingProperties)) {
+            try (final InputStream in = Files.newInputStream(loggingProperties)) {
+                System.setProperty(LOG_BOOT_FILE_PROP, bootLog.toAbsolutePath().toString());
+                PropertyConfigurator configurator = new PropertyConfigurator(embeddedLogContext);
+                configurator.configure(in);
+            }
+        }
+        return embeddedLogContext;
+    }
+
+    public void run() throws Exception {
+        try {
+            server = buildServer(startServerArgs);
+        } catch (RuntimeException ex) {
+            cleanup();
+            throw ex;
+        }
+
+        Runtime.getRuntime().addShutdownHook(new ShutdownHook());
+        server.start();
+    }
+
+    private void cleanup() {
+        log.deletingHome(jbossHome);
+        deleteDir(jbossHome);
+
+    }
+
+    private Server buildServer(List<String> args) throws IOException {
+        String[] array = new String[args.size()];
+        log.advertiseOptions(args);
+        return Server.newSever(jbossHome, args.toArray(array), loader, this);
+    }
+
+    private static void deleteDir(Path root) {
+        if (root == null || !Files.exists(root)) {
+            return;
+        }
+        try {
+            Files.walkFileTree(root, new SimpleFileVisitor<Path>() {
+                @Override
+                public FileVisitResult visitFile(Path file, BasicFileAttributes attrs)
+                        throws IOException {
+                    try {
+                        Files.delete(file);
+                    } catch (IOException ex) {
+                    }
+                    return FileVisitResult.CONTINUE;
+                }
+
+                @Override
+                public FileVisitResult postVisitDirectory(Path dir, IOException e)
+                        throws IOException {
+                    if (e != null) {
+                        // directory iteration failed
+                        throw e;
```

This seems odd as we swallow the error below at line 291.

---

Yep, need to remove it.